runtime/binding-mqtt/src/main/zilla/protocol.idl

```diff
@@ -22,20 +22,34 @@
         enum MqttPacketType
         {
             RESERVED,
-            CONNECT,
-            CONNACK,
-            PUBLISH,
-            PUBACK,
-            PUBREC,
-            PUBREL,
-            PUBCOMP,
-            SUBSCRIBE,
-            SUBACK,
-            UNSUBSCRIBE,
-            UNSUBACK,
+            CONNECTV4,
+            CONNECTV5,
+            CONNACKV4,
+            CONNACKV5,
+            PUBLISHV4,
+            PUBLISHV5,
+            PUBACKV4,
+            PUBACKV5,
+            PUBRECV4,
+            PUBRECV5,
+            PUBRELV4,
+            PUBRELV5,
+            PUBCOMPV4,
+            PUBCOMPV5,
+            SUBSCRIBEV4,
+            SUBSCRIBEV5,
+            SUBACKV4,
+            SUBACKV5,
+            UNSUBSCRIBEV4,
+            UNSUBSCRIBEV5,
+            UNSUBACK4,
+            UNSUBACKV5,
             PINGREQ,
+            PINGREQ,
+            PINGRESP,
             PINGRESP,
-            DISCONNECT,
+            DISCONNECTV4,
+            DISCONNECTV5,
```

Comment:
I wasn't expecting this change, we only need the super set of all packet types as they have the same values across protocols, right?

Then the decoders would be registered separately for v4 vs v5 instead of trying to put them all in the same map, agree?